### ISSUE TITLE
Failed to execute: /usr/bin/perl /opt/ETP/bin/gmes//gmes_petap.pl

### ISSUE THREAD

**Issue by mabh5** [Gaius-Augustus/BRAKER#677]
Hello,

I see that this error shows up in the issues list quite a bit. In my case, I am running braker3 using a singularity image built and tested today using these instructions: [Braker3 Install](https://hub.docker.com/r/teambraker/braker3)

I also downloaded the .gm_key and put it in my home directory as instructed. The key is new and so isn't expired. I checked that singularity can access the key with singularity shell:

```
$ srun --pty singularity shell --writable-tmpfs -B ${PWD}:${PWD} braker3.sif
srun: Account not specified in script or ~/.default_slurm_account, using latest project
To request GPUs, add --gpus-per-node X or --gpus X, where X is the desired number of GPUs.
Job 2067302 running on hopper002
Singularity> ls .gm_key
.gm_key
```

The command passed to the singularity image was ```braker.pl --genome=corrected_scaffoldsmasked.fasta --prot_seq=proteinsvert.fasta``` 

the protein sequences were downloaded from OrthoDB as instructed. 

```
Failed to execute: /usr/bin/perl /opt/ETP/bin/gmes//gmes_petap.pl --verbose --seq /users/mh/braker/genome.fa --EP /users/mh/braker/genemark_hintsfile.gff --cores=1  --gc_donor 0.001 --evidence /users/mh/braker/genemark_evidence.gff  --soft_mask auto 1>/users/mh/braker/GeneMark-EP.stdout 2>/users/mh/braker/errors/GeneMark-EP.stderr
The most common problem is an expired or not present file ~/.gm_key!
```

Any assistance or advice would be greatly appreciated.

**Comment by kmkocot** [comment]
Providing any non-empty files from the errors directory might be helpful to the admins.

**Comment by KatharinaHoff** [comment]
Non-empty error files are always helpful, but please request help on this
from the GeneMark developers via e-mail. I am not sure whether they have
the issues on Github on their radar.

Kevin M. Kocot ***@***.***> schrieb am Fr. 6. Okt. 2023 um
15:15:

**Comment by alexlomsadze** [comment]
Hello,

First, I want to comment on reoccurring reports about the GeneMark key file.
How do you ensure that the key file for GeneMark is correctly installed?

Whenever there is a concern about the key file, find the GeneMak.hmm executable and check if the GeneMark.hmm code complains about the key:
```
find path_to_genemark_folder  -name gmhmme3
path_to/gmhmme3 -v
```
In the case of the BRAKER singularity container:
```
singularity shell braker3.sif
find /opt  -name gmhmme3
/opt/ETP/bin/gmes/gmhmme3  -v
exit
```

If GeneMark.hmm complains about the key, then yes, there is a problem with it.
If GeneMark.hmm complains about the missing input file with sequence, then you are set.

Alex

**Comment by alexlomsadze** [comment]
Hello,

We may start with two logfiles:
  /users/mh/braker/GeneMark-EP.stdout
  /users/mh/braker/errors/GeneMark-EP.stderr

My email is alexl@gatech.edu

Another option is to open an issue at GeneMark-ETP repository 
https://github.com/gatech-genemark/GeneMark-ETP
or we can continue here.

In any case, logfiles are a starting point.

Alex

**Comment by KatharinaHoff** [comment]
The monolithic braker.pl Perl script has been replaced entirely in [BRAKER4](https://github.com/Gaius-Augustus/BRAKER4). The pipeline is now a Snakemake workflow where each step is a separate rule running inside a Singularity container. All file path management, directory creation, and tool invocation are handled by Snakemake. This class of braker.pl errors no longer exists. Please try BRAKER4 and report any issues there.
